## 목차
1. 필요 라이브러리 불러오기<br><br>

2. 데이터셋 불러오기<br><br>

3. 데이터 확인<br>
    3-1. 형태, features, 크기 파악<br><br>
    
4. 데이터 전처리<br>
    4-1. 각 feature 파악/탐색<br>
    4-2. 불필요한 feature 삭제<br>
    4-3. feature 결합<br>
    4-4. 전처리 데이터 확인<br><br>
    
5. 데이터 저장<br><br>

6. 요약

## 1. 필요 라이브러리 불러오기

In [ ]:
import pandas as pd
import numpy as np

+ 하기 사용 csv 파일 엑셀로 열었을 시, 글자가 깨지는 현상 발생  
+ 원인: 한글 인코딩 후, 디코딩 시 다른 규격 적용
    - eg) csv 파일 생성 시, utf-8 규격 인코딩 진행 -> 엑셀 상 csv 열 때, 949: 한국어 ANSI 코드 디코딩 설정 -> 한글 깨짐  
+ 해결:  
    1. 윈도우 메모장 사용  
    2. 메모장에서 해당 cvs 열기  
    3. 다른 이름으로 저장; 하단 인코딩: ANSI 선택  
    4. 저장  

## 2. 데이터셋 불러오기

In [ ]:
comm = pd.read_csv('C:/Python/01 Group Works/Project_InnoPost/Received Data/07_상권정보(대전)/소상공인시장진흥공단_상가(상권)정보_20220930/소상공인시장진흥공단_상가(상권)정보_대전_202209.csv')

## 3. 데이터 확인

In [ ]:
# pd.options.display.max_rows = 6
    # 추후 보여질 데이터프레임의 열 개수 지정

In [ ]:
comm

상가업소번호         상호명    지점명 상권업종대분류코드 상권업종대분류명 상권업종중분류코드   상권업종중분류명  \
0      22401931      스피드메이트    NaN         F    생활서비스       F14    자동차/이륜차   
1      24185362  국제종합기계부품총판    NaN         D       소매       D25      기타판매업   
2      22742209    알파수학전문학원    NaN         R    학문/교육       R01  학원-보습교습입시   
3      17174039         솔바람    NaN         Q       음식       Q01         한식   
4      17174056       BBQ치킨  대전대흥점         Q       음식       Q09       유흥주점   
...         ...         ...    ...       ...      ...       ...        ...   
75140  18774117        마을세탁     본점         F    생활서비스       F02   세탁/가사서비스   
75141  18754136        뉴로아이    NaN         D       소매       D02  선물/팬시/기념품   
75142  18771449         플랜비    NaN         Q       음식       Q12     커피점/카페   
75143  18756774        만복식당    NaN         Q       음식       Q01         한식   
75144  18764726       수진네야채    NaN         D       소매       D25      기타판매업   

      상권업종소분류코드     상권업종소분류명 표준산업분류코드  ...                     건물관리번호  \
0        F14A01    자동차정비/카센타      NaN  ...  3020014600111220000001772   
1        D25A15        농기구판매   G47512  ...  3017010300101730001030969   
2        R01A01        학원-입시   P85501  ...  3014010200101570008028501   
3        Q01A01    한식/백반/한정식   I56111  ...  3020012000109080016009529   
4        Q09A01        호프/맥주   I56219  ...  3014010500101180005012316   
...         ...          ...      ...  ...                        ...   
75140    F02A01      세탁소/빨래방   S96912  ...  3023011000101060003016055   
75141    D02A02       꽃집/꽃배달   G47851  ...  3017011600115580010000001   
75142    Q12A01  커피전문점/카페/다방   I56220  ...  3017011600111180000018357   
75143    Q01A01    한식/백반/한정식   I56111  ...  3023010100100440001015026   
75144    D25A03    노점/유사이동판매   G47920  ...  3011010600102120000029783   

            건물명                           도로명주소   구우편번호  신우편번호  동정보  층정보  호정보  \
0           NaN        대전광역시 유성구 관들4길 25, (관평동)  305509  34018  NaN  NaN  NaN   
1          써머스빌  대전광역시 서구 계백로 1426, (도마동, 써머스빌)  302823  35400  NaN  NaN  NaN   
2          선우빌딩      대전광역시 중구 선화서로 128-1, (선화동)  301824  34841  NaN    2  NaN   
3           NaN  대전광역시 유성구 은구비서로3번길 8-23, (지족동)  305330  34086  NaN  NaN  NaN   
4           NaN         대전광역시 중구 대흥로 158, (대흥동)  301802  34928  NaN    1  NaN   
...         ...                             ...     ...    ...  ...  ...  ...   
75140       NaN    대전광역시 대덕구 우암로491번길 43, (비래동)  306808  34417  NaN    1  NaN   
75141       NaN  대전광역시 서구 관저남로25번길 23-82, (관저동)  302812  35383  NaN  NaN  NaN   
75142       NaN     대전광역시 서구 관저중로95번길 90, (관저동)  302812  35372  NaN    1  NaN   
75143  농수산물도매시장       대전광역시 대덕구 한밭대로 987, (오정동)  306817  34377  NaN    2  NaN   
75144   신흥마을아파트  대전광역시 동구 옥천로 38, (신흥동, 신흥마을주공)  300700  34669  105  NaN  NaN   

               경도         위도  
0      127.393241  36.424448  
1      127.379765  36.312625  
2      127.413968  36.334319  
3      127.320105  36.373504  
4      127.429579  36.324517  
...           ...        ...  
75140  127.446986  36.357756  
75141  127.336851  36.295072  
75142  127.333695  36.300940  
75143  127.406460  36.357823  
75144  127.446082  36.320079  

[75145 rows x 39 columns]

In [ ]:
comm.info()
    # 데이터셋 정보 확인

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 75145 entries, 0 to 75144
Data columns (total 39 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   상가업소번호     75145 non-null  int64  
 1   상호명        75145 non-null  object 
 2   지점명        11894 non-null  object 
 3   상권업종대분류코드  75145 non-null  object 
 4   상권업종대분류명   75145 non-null  object 
 5   상권업종중분류코드  75145 non-null  object 
 6   상권업종중분류명   75145 non-null  object 
 7   상권업종소분류코드  75145 non-null  object 
 8   상권업종소분류명   75145 non-null  object 
 9   표준산업분류코드   69614 non-null  object 
 10  표준산업분류명    69614 non-null  object 
 11  시도코드       75145 non-null  int64  
 12  시도명        75145 non-null  object 
 13  시군구코드      75145 non-null  int64  
 14  시군구명       75145 non-null  object 
 15  행정동코드      75145 non-null  int64  
 16  행정동명       75145 non-null  object 
 17  법정동코드      75145 non-null  int64  
 18  법정동명       75145 non-null  object 
 19  지번코드       75145 non-null  int64  
 20  대지구분코드

In [ ]:
%%html
<style>
table {float: left}
</style>

## 4. 데이터 전처리
### 4-1. 각 feature 탐색

| #   	| COL NAME           	| USE 	| NOTES         	|
|:-----	|:--------------------	|:-----:|:---------------	|
| 1.  	| 상가업소번호       	| X   	|               	|
| 2.  	| 상호명             	|     	| 지점명과 결합; 상호명만 명시할 시, 타 지역 상권과 혼동 가능성 고려|
| 3.  	| 지점명             	|     	| 상단 참고     	|
| 4.  	| 상권업종대분류코드 	| X   	|               	|
| 5.  	| 상권업종대분류명   	|     	|흡연지수 산출 목적|
| 6.  	| 상권업종중분류코드 	| X   	|               	|
| 7.  	| 상권업종중분류명   	|     	|흡연지수 산출 목적|
| 8.  	| 상권업종소분류코드 	| X   	|               	|
| 9.  	| 상권업종소분류명   	|     	|흡연지수 산출 목적|
| 10. 	| 표준산업분류코드   	| X   	|               	|
| 11. 	| 표준산업분류명     	| X    	|               	|
| 12. 	| 시도코드           	| X   	|               	|
| 13. 	| 시도명             	|     	|지역단위 별 상권수 등 계산 목적|
| 14. 	| 시군구코드         	| X   	|               	|
| 15. 	| 시군구명           	|     	|지역단위 별 상권수 등 계산 목적|
| 16. 	| 행정동코드         	| X   	|               	|
| 17. 	| 행정동명           	|     	|지역단위 별 상권수 등 계산 목적|
| 18. 	| 법정동코드         	| X   	|               	|
| 19. 	| 법정동명           	| X   	|지역단위 별 상권수 등 계산 목적|
| 20. 	| 지번코드           	| X   	|               	|
| 21. 	| 대지구분코드       	| X   	|               	|
| 22. 	| 대지구분명         	| X   	|               	|
| 23. 	| 지번본번지         	| X   	|               	|
| 24. 	| 지번부번지         	| X   	|               	|
| 25. 	| 지번주소           	|X     	|               	|
| 26. 	| 도로명코드         	| X   	|               	|
| 27. 	| 도로명             	| X   	|               	|
| 28. 	| 건물본번지         	| X   	|               	|
| 29. 	| 건물부번지         	| X   	|               	|
| 30. 	| 건물관리번호       	| X   	|               	|
| 31. 	| 건물명             	| X   	|               	|
| 32. 	| 도로명주소         	|     	|각 상권의 상세주소로서 참고|
| 33. 	| 구우편번호         	| X   	|각 상권의 상세주소로서 참고|
| 34. 	| 신우편번호         	| X   	|각 상권의 상세주소로서 참고|
| 35. 	| 동정보             	| X   	|               	|
| 36. 	| 층정보             	| X   	|               	|
| 37. 	| 호정보             	| X   	|               	|
| 38. 	| 경도               	|     	| 시각화 목적|
| 39. 	| 위도               	|     	| 시각화 목적|

총 39개 열 중,
- 흡연지수 도출 및 QGIS 시각화 시, 불필요한 열 삭제 예정; (25)
    1. 각 상권, 지역에 부여된 식별 번호/코드 + 표준산업분류명 (6) 
    2. 지역/주소관련 코드 및 상세/중복 정보 삭제 (22)<br><br>
- 사용 가능 정보; (14)
    1. 상호명 + 지점명 (2)
    2. 상권업종대분류명
    3. 상권업종중분류명
    4. 상권업종소분류명
    5. 시도명
    6. 시군구명
    7. 행정동명
    8. 법정동명
    9. 도로명주소
    10. 구우편번호
    11. 신우편번호
    12. 경도
    13. 위도

### 4-2. 불필요한 feature 삭제

In [ ]:
comm.drop(['상가업소번호', '상권업종대분류코드', '상권업종중분류코드', '상권업종소분류코드', '표준산업분류코드', '표준산업분류명', '시도코드', '시군구코드', '행정동코드', '법정동코드', '지번코드', '대지구분코드', '대지구분명', '지번본번지', '지번부번지', '지번주소', '도로명코드', '도로명', '건물본번지', '건물부번지', '건물관리번호', '건물명', '동정보', '층정보', '호정보'], axis = 'columns', inplace = True)
    # 열 삭제 진행
    # axis = 'columns' == axis = 1; 열 단위 삭제
    # inplace = True; 변경사항 저장

### 4-3. feature 결합

In [ ]:
comm['지점명'].fillna('', inplace = True)
    # NaN --> 공란 변경, 저장

comm['상호-지점명'] = comm['상호명'] + ' ' + comm['지점명']
    # 상호명 + 지점명 결합 --> 새로운 열, 상호-지점명 생성

In [ ]:
comm.drop(['상호명', '지점명'], axis = 1, inplace = True)

In [ ]:
comm = comm[['상호-지점명', '상권업종대분류명', '상권업종중분류명', '상권업종소분류명', '시도명', '시군구명', '행정동명', '법정동명', '도로명주소', '구우편번호', '신우편번호', '경도', '위도']]

In [ ]:
comm

상호+지점명 상권업종대분류명   상권업종중분류명     상권업종소분류명    시도명 시군구명   행정동명 법정동명  \
0          스피드메이트     생활서비스    자동차/이륜차    자동차정비/카센타  대전광역시  유성구    관평동  관평동   
1      국제종합기계부품총판        소매      기타판매업        농기구판매  대전광역시   서구   도마2동  도마동   
2        알파수학전문학원     학문/교육  학원-보습교습입시        학원-입시  대전광역시   중구  은행선화동  선화동   
3             솔바람        음식         한식    한식/백반/한정식  대전광역시  유성구   노은1동  지족동   
4      BBQ치킨 대전대흥점       음식       유흥주점        호프/맥주  대전광역시   중구    대흥동  대흥동   
...            ...      ...        ...          ...    ...  ...    ...  ...   
75140      마을세탁 본점    생활서비스   세탁/가사서비스      세탁소/빨래방  대전광역시  대덕구    비래동  비래동   
75141        뉴로아이        소매  선물/팬시/기념품       꽃집/꽃배달  대전광역시   서구   관저2동  관저동   
75142         플랜비        음식     커피점/카페  커피전문점/카페/다방  대전광역시   서구   관저2동  관저동   
75143        만복식당        음식         한식    한식/백반/한정식  대전광역시  대덕구    오정동  오정동   
75144       수진네야채        소매      기타판매업    노점/유사이동판매  대전광역시   동구    신인동  신흥동   

                                도로명주소   구우편번호  신우편번호          경도         위도  
0            대전광역시 유성구 관들4길 25, (관평동)  305509  34018  127.393241  36.424448  
1      대전광역시 서구 계백로 1426, (도마동, 써머스빌)  302823  35400  127.379765  36.312625  
2          대전광역시 중구 선화서로 128-1, (선화동)  301824  34841  127.413968  36.334319  
3      대전광역시 유성구 은구비서로3번길 8-23, (지족동)  305330  34086  127.320105  36.373504  
4             대전광역시 중구 대흥로 158, (대흥동)  301802  34928  127.429579  36.324517  
...                               ...     ...    ...         ...        ...  
75140    대전광역시 대덕구 우암로491번길 43, (비래동)  306808  34417  127.446986  36.357756  
75141  대전광역시 서구 관저남로25번길 23-82, (관저동)  302812  35383  127.336851  36.295072  
75142     대전광역시 서구 관저중로95번길 90, (관저동)  302812  35372  127.333695  36.300940  
75143       대전광역시 대덕구 한밭대로 987, (오정동)  306817  34377  127.406460  36.357823  
75144  대전광역시 동구 옥천로 38, (신흥동, 신흥마을주공)  300700  34669  127.446082  36.320079  

[75145 rows x 13 columns]

### 4-4. 전처리 데이터 확인

In [ ]:
comm.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 75145 entries, 0 to 75144
Data columns (total 13 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   상호+지점명    75145 non-null  object 
 1   상권업종대분류명  75145 non-null  object 
 2   상권업종중분류명  75145 non-null  object 
 3   상권업종소분류명  75145 non-null  object 
 4   시도명       75145 non-null  object 
 5   시군구명      75145 non-null  object 
 6   행정동명      75145 non-null  object 
 7   법정동명      75145 non-null  object 
 8   도로명주소     75145 non-null  object 
 9   구우편번호     75145 non-null  int64  
 10  신우편번호     75145 non-null  int64  
 11  경도        75145 non-null  float64
 12  위도        75145 non-null  float64
dtypes: float64(2), int64(2), object(9)
memory usage: 7.5+ MB


In [ ]:
comm[['상호-지점명', '상권업종대분류명', '상권업종중분류명', '상권업종소분류명', '시도명', '시군구명', '행정동명', '법정동명', '도로명주소']].describe()

상호+지점명 상권업종대분류명 상권업종중분류명   상권업종소분류명    시도명   시군구명   행정동명   법정동명  \
count   75145    75145    75145      75145  75145  75145  75145  75145   
unique  67276        8       88        581      1      5     82    166   
top      부동산        소매       한식  한식/백반/한정식  대전광역시     서구   온천1동    둔산동   
freq       45    24924     9140       6346  75145  23818   2850   4139   

                          도로명주소  
count                     75145  
unique                    30384  
top     대전광역시 서구 계룡로 598, (괴정동)  
freq                        423

In [ ]:
comm.to_csv('데이터 정제_상권정보_mod.csv', index = False, encoding='cp949')

In [ ]:
pd.set_option('display.max_rows', None)

comm_mod = pd.pivot_table(comm,
                          index = ['상권업종대분류명', '상권업종중분류명'],
                          columns = '시군구명',
                          values = '상호-지점명',
                          aggfunc = 'count')
    # 피벗테이블 생성

comm_mod

**주요 흡연 지역 및 [금연지역](https://easylaw.go.kr/CSP/CnpClsMain.laf?popMenu=ov&csmSeq=908&ccfNo=4&cciNo=1&cnpClsNo=1) 관련 상권업종대분류명 선정하여 필터**

In [ ]:
comm_mod.query("상권업종대분류명 == ['관광/여가/오락', '생활서비스', '소매', '숙박', '음식', '학문/교육']")
    # 피벗 테이블 상 특정 열 필터

## 5. 요약


### 데이터 전처리 진행 상황
+ 데이터 축소
    - 75145 rows × 39 columns --> 75145 rows × 13 columns
    - 22.4+ MB --> 5.7+ MB
    - 최종 데이터; string & float type, non-null 값 구성<br><br>
    
+ 데이터 열 별 사용목적 식별 (총 10개 열)
    1. (1) 상호-지점명: count
    2. (3) 상권업종대분류명 / 상권업종중분류명 / 상권업종소분류명: 흡연지수 산출 시 사용
    3. (3) 시도명 / 시군구명 / 행정동명 / 법정동명: 지역단위 별 상권수 등 계산 목적
    4. (1) 도로명주소 / 구우편번호 / 신우편번호: 각 상권의 상세주소로서 참고
    5. (2) 경도 / 위도: 시각화 목적

### 추가 고려 필요한 사항
+ 흡연 위험 지역 정의
    - 금연지역임에도 흡연(민원 등) 빈번하게 발생하는 지역
    - 흡연 가능 구역이나 흡연 취약 계층 유동인구가 많은 지역 등<br><br>
    
+ 흡연 취약 계층 정의
    - 미성년자 (만 0-19세)
    - 임산부
    - 비흡연 노인 등 
    - **인구 파악 가능여부 확인 필요**<br><br>
    
+ 폐업장 (2022년 자료 다운로드 완료) 등

### 추후 기타 참고 자료
+ [지역 별 건강 프로파일](https://chs.kdca.go.kr/chs/recsRoom/areaResultMain.do#)
    - 건강행태에 대한 설명 - 비흡연자의 공공장소 간접흡연 노출률 (2020); 서구 > 대덕구 > 중구 > 동구 > 유성구<br><br>
    
+ [지역사회 건강관련 요인 데이터베이스](https://chs.kdca.go.kr/chs/recsRoom/dataBaseMain.do)
    - 대분류: 건강행태
    - 중분류: 흡연
    - 소분류: 간접흡연, 흡연, 금연 등<br><br>
    
+ [기사1](https://www.chungnamilbo.co.kr/news/articleView.html?idxno=673179)
    - 대전 도심 '재떨이화' ...간접흡연에 비흡연자 몸살, 흡연공간 부족...번화가 길바닥 꽁초 가득 (2022.08)<br><br>
    
+ [기사2](http://www.daejonilbo.com/news/articleView.html?idxno=1483063)
    - 대전 금연거리/공원 '무용지물'...보건당국 코로나 방역 업무 지원에 단속 한계 (2021.08)<br><br>